File: logic/src/test/java/by/rekuts/travelagency/logic/service/integration/TourServiceImplTest.java
//package by.rekuts.travelagency.logic.service.integration;
//
//import by.rekuts.travelagency.logic.config.TestRepositoryConfig;
//import by.rekuts.travelagency.logic.repository.CountrySpecification;
//import by.rekuts.travelagency.logic.repository.HotelSpecification;
//import by.rekuts.travelagency.logic.repository.TourSpecification;
//import by.rekuts.travelagency.logic.domain.Tour;
//import by.rekuts.travelagency.logic.service.CountryService;
//import by.rekuts.travelagency.logic.service.HotelService;
//import by.rekuts.travelagency.logic.service.TourService;
//import org.junit.Before;
//import org.junit.Test;
//import org.junit.runner.RunWith;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.test.context.ActiveProfiles;
//import org.springframework.test.context.ContextConfiguration;
//import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
//import org.springframework.transaction.annotation.Transactional;
//
//import javax.persistence.PersistenceException;
//import java.math.BigDecimal;
//import java.time.LocalDate;
//import java.util.List;
//
//import static org.junit.Assert.assertEquals;
//import static org.junit.Assert.assertTrue;
//
//@RunWith(SpringJUnit4ClassRunner.class)
//@ContextConfiguration(classes = TestRepositoryConfig.class)
//@ActiveProfiles("testScope")
//@Transactional
//public class TourServiceImplTest {
//    @Autowired
//    TourService tourService;
//    @Autowired
//    HotelService hotelService;
//    @Autowired
//    CountryService countryService;
//
//    private Tour tour = new Tour();
//
//    @Before
//    public void initializeToursList() {
//        tour.setPhoto("photo.jpg");
//        tour.setDate(LocalDate.now());
//        tour.setDuration(14);
//        tour.setDescription("Interesting tour1");
//        tour.setCost(BigDecimal.valueOf(900));
//        tour.setTourType(Tour.TourType.cultural);
//        tour.setHotel(hotelService.getList(new HotelSpecification(68)).get(0));
//        tour.setCountry(countryService.getList(new CountrySpecification(91)).get(0));
//    }
//
//    @Test
//    public void insertTest(){
//        int countToursFirst = tourService.getList(new TourSpecification()).size();
//        tourService.insert(tour);
//        int countToursLast = tourService.getList(new TourSpecification()).size();
//        assertEquals(1, countToursLast - countToursFirst);
//    }
//
//    @Test (expected = PersistenceException.class)
//    public void insertTestFalse(){
//        tour.setId(1);
//        tourService.insert(tour);
//    }
//
//    @Test
//    public void deleteTest() {
//        int countToursFirst = tourService.getList(new TourSpecification()).size();
//        tourService.delete(1);
//        int countToursLast = tourService.getList(new TourSpecification()).size();
//        assertEquals(1, countToursFirst - countToursLast);
//    }
//
//    @Test
//    public void getTourByIdTestTrue() {
//        Tour tour = tourService.getList(new TourSpecification(4)).get(0);
//        assertEquals("Nam dui.", tour.getDescription());
//    }
//
//    @Test (expected = IndexOutOfBoundsException.class)
//    public void getTourByIdTestFalse1() {
//        tourService.getList(new TourSpecification(1001)).get(0);
//    }
//
//    @Test
//    public void getTourByIdTestFalse2() {
//        List<Tour> tours = tourService.getList(new TourSpecification(1001));
//        assertTrue(tours.isEmpty());
//    }
//
//    @Test
//    public void getAllToursTest() {
//        List<Tour> tourList = tourService.getList(new TourSpecification());
//        assertEquals(1000, tourList.size());
//    }
//
//    @Test
//    public void getToursByCriteria() {
//        TourSpecification specification = new TourSpecification();
//        specification.setTourType(Tour.TourType.safari);
//        List<Tour> result = tourService.getList(specification);
//        assertEquals(58, result.size());
//    }
//
//    @Test
//    public void getToursByUserId() {
//        TourSpecification specification = new TourSpecification();
//        specification.setUserId(101);
//        List<Tour> result = tourService.getList(specification);
//        assertEquals(2, result.size());
//    }
//}

File: service/src/test/scala/by/rekuts/travelagency/service/integration/TourServiceImplTest.scala
package by.rekuts.travelagency.service.integration

import java.math.BigDecimal
import java.time.LocalDate

import by.rekuts.travelagency.domain.{Tour, TourType}
import by.rekuts.travelagency.repository.config.TestRepositoryConfig
import by.rekuts.travelagency.repository.{CountrySpecification, HotelSpecification, TourSpecification}
import by.rekuts.travelagency.service.{CountryService, HotelService, TourService}
import javax.persistence.PersistenceException
import org.junit.Assert.{assertEquals, assertTrue}
import org.junit.runner.RunWith
import org.junit.{Before, Test}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
import org.springframework.test.context.{ActiveProfiles, ContextConfiguration}
import org.springframework.transaction.annotation.Transactional

@RunWith(classOf[SpringJUnit4ClassRunner])
@ContextConfiguration(classes = Array(classOf[TestRepositoryConfig]))
@ActiveProfiles(Array("testScope"))
@Transactional
class TourServiceImplTest {
  @Autowired
  private var tourService: TourService = _
  @Autowired
  private var hotelService: HotelService = _
  @Autowired
  private var countryService: CountryService = _

  private val tour = new Tour

  @Before def initializeToursList(): Unit = {
    tour.photo_$eq("photo.jpg")
    tour.date_$eq(LocalDate.now)
    tour.duration_$eq(14)
    tour.description_$eq("Interesting tour1")
    tour.cost_$eq(BigDecimal.valueOf(900))
    tour.tourType_$eq(TourType.cultural)
    tour.hotel_$eq(hotelService.getList(new HotelSpecification(68)).get(0))
    tour.country_$eq(countryService.getList(new CountrySpecification(91)).get(0))
  }

  @Test
  def insertTest(): Unit = {
    val countToursFirst = tourService.getList(new TourSpecification).size
    tourService.insert(tour)
    val countToursLast = tourService.getList(new TourSpecification).size
    assertEquals(1, countToursLast - countToursFirst)
  }

  @Test(expected = classOf[PersistenceException])
  def insertTestFalse(): Unit = tourService.insert(new Tour)

  @Test
  def deleteTest(): Unit = {
    val countToursFirst = tourService.getList(new TourSpecification).size
    tourService.delete(1)
    val countToursLast = tourService.getList(new TourSpecification).size
    assertEquals(1, countToursFirst - countToursLast)
  }

  @Test
  def obtainTourByIdTestTrue(): Unit = {
    val tour = tourService.getList(new TourSpecification(4)).get(0)
    assertEquals("Nam dui.", tour.description)
  }

  @Test(expected = classOf[IndexOutOfBoundsException])
  def obtainTourByIdTestFalse1(): Unit = {
    tourService.getList(new TourSpecification(1001)).get(0)
  }

  @Test
  def obtainTourByIdTestFalse2(): Unit = {
    val tours = tourService.getList(new TourSpecification(1001))
    assertTrue(tours.isEmpty)
  }

  @Test
  def obtainAllToursTest(): Unit = {
    val tourList = tourService.getList(new TourSpecification)
    assertEquals(1002, tourList.size)
  }

  @Test
  def obtainToursByCriteria(): Unit = {
    val specification = new TourSpecification
    specification.tourType_$eq(TourType.safari)
    val result = tourService.getList(specification)
    assertEquals(58, result.size)
  }

  @Test
  def obtainToursByUserId(): Unit = {
    val specification = new TourSpecification
    specification.userId_$eq(101)
    val result = tourService.getList(specification)
    assertEquals(2, result.size)
  }
}
